Sources/AsyncHTTPClient/Utils.swift

```diff
@@ -52,174 +53,198 @@ public final class HTTPClientCopyingDelegate: HTTPClientResponseDelegate {
+                        }
+                        return channel.eventLoop.makeSucceededVoidFuture()
+                    } catch {
+                        return channel.eventLoop.makeFailedFuture(error)
                     }
-                    return channel.eventLoop.makeSucceededVoidFuture()
-                } catch {
-                    return channel.eventLoop.makeFailedFuture(error)
                 }
             }
     }
 
-    static func makeHTTP1Channel(destination: ConnectionPool.Key, eventLoop: EventLoop, configuration: HTTPClient.Configuration, preference: HTTPClient.EventLoopPreference) -> EventLoopFuture<Channel> {
-        let channelEventLoop = preference.bestEventLoop ?? eventLoop
+    /// Creates the best-suited bootstrap given an `EventLoop` and pairs it with an appropriate TLS provider.
+    private static func makeBestBootstrap(
+        host: String,
+        eventLoop: EventLoop,
+        requiresTLS: Bool,
+        sslContextCache: SSLContextCache,
+        tlsConfiguration: TLSConfiguration,
+        useProxy: Bool,
+        logger: Logger
+    ) -> EventLoopFuture<NIOClientTCPBootstrap> {
+        #if canImport(Network)
+            // if eventLoop is compatible with NIOTransportServices create a NIOTSConnectionBootstrap
+            if #available(OSX 10.14, iOS 12.0, tvOS 12.0, watchOS 6.0, *), let tsBootstrap = NIOTSConnectionBootstrap(validatingGroup: eventLoop) {
+                // create NIOClientTCPBootstrap with NIOTS TLS provider
+                let parameters = tlsConfiguration.getNWProtocolTLSOptions()
+                let tlsProvider = NIOTSClientTLSProvider(tlsOptions: parameters)
+                return eventLoop.makeSucceededFuture(NIOClientTCPBootstrap(tsBootstrap, tls: tlsProvider))
+            }
+        #endif
 
-        let key = destination
+        if let clientBootstrap = ClientBootstrap(validatingGroup: eventLoop) {
+            // If there is a proxy don't create TLS provider as it will be added at a later point.
+            if !requiresTLS || useProxy {
+                return eventLoop.makeSucceededFuture(NIOClientTCPBootstrap(clientBootstrap,
+                                                                           tls: NIOInsecureNoTLS()))
+            } else {
+                return sslContextCache.sslContext(tlsConfiguration: tlsConfiguration,
+                                                  eventLoop: eventLoop,
+                                                  logger: logger)
+                    .flatMapThrowing { sslContext in
+                        let hostname = (host.isIPAddress || host.isEmpty) ? nil : host
+                        let tlsProvider = try NIOSSLClientTLSProvider<ClientBootstrap>(context: sslContext, serverHostname: hostname)
+                        return NIOClientTCPBootstrap(clientBootstrap, tls: tlsProvider)
+                    }
```

Comment:
Just ran the tests on macOS and noticed that this code path is never hit.

Reply:
@fabianfett yes, expected. macOS is using NIOTS.

Reply:
and thanks, added a test case which will trigger this code even on macOS so we don't rely on Linux.